packages/notte-core/src/notte_core/browser/dom_tree.py

```diff
@@ -431,7 +448,14 @@ def prune_non_dialogs_if_present(self) -> Sequence["DomNode"]:
         """TODO: make it work with A11yNode and DomNode"""
 
         def is_dialog(node: DomNode) -> bool:
-            return node.role == NodeRole.DIALOG and node.computed_attributes.in_viewport
+            if node.role != NodeRole.DIALOG:
+                return False
+            if node.computed_attributes.in_viewport is False:
+                return False
+            if len(node.interaction_nodes()) == 0:
```

Comment:
FYI this was a bug. We don't want to only highlight the dialog if it does not contain any interaction node (the agent will be stuck otherwise)